libsolidity/codegen/CompilerContext.cpp

```diff
@@ -298,12 +298,14 @@ unsigned CompilerContext::baseStackOffsetOfVariable(Declaration const& _declarat
 
 unsigned CompilerContext::baseToCurrentStackOffset(unsigned _baseOffset) const
 {
-	return m_asm->deposit() - _baseOffset - 1;
+	solAssert(static_cast<unsigned>(m_asm->deposit()) > _baseOffset, "");
+	return static_cast<unsigned>(m_asm->deposit()) - _baseOffset - 1;
 }
 
 unsigned CompilerContext::currentToBaseStackOffset(unsigned _offset) const
 {
-	return m_asm->deposit() - _offset - 1;
+	solAssert(static_cast<unsigned>(m_asm->deposit()) > _offset, "");
```

Comment:
object "<RuntimeObject>" {
				code {
					<memoryInit>
					<dispatch>
					<runtimeFunctions>
				}
				<runtimeSubObjects>
			}

Reply:
`_baseOffset` >= `m_asm->deposit()` would be a bug, right?

The assertion wasn't there before but I suggested adding it since otherwise this might wrap around. But I don't know this code well. There's a small chance that this was the intention.

> we did even consider trying to actively get rid of the "" assertions at some point :-) )

Wouldn't really be that hard to add a variant without the message and gradually keep replacing them. Do we actually want this? I thought it was by design, to force you to add a message.